packages/hash/frontend/src/components/layout/PageHeader/SearchBar/SearchInput.tsx

```diff
@@ -0,0 +1,205 @@
+      </Tooltip>
+    </Box>
+  );
+};
+
+const ShortcutIcon = () => {
+  const isMac = navigator.userAgent.toUpperCase().includes("MAC");
+
+  return (
+    <Box
+      sx={(theme) => ({
+        marginRight: theme.spacing(1),
+        height: "100%",
+        pointerEvents: "none",
+        display: "flex",
+        alignItems: "center",
+        fontWeight: "bold",
+      })}
+    >
+      <Box
+        sx={(theme) => ({
+          height: "26px",
+          display: "flex",
+          alignItems: "center",
+          justifyContent: "center",
+          color: theme.palette.gray[50],
+          backgroundColor: theme.palette.gray[20],
+          borderRadius: "2px",
+          px: 1,
+          py: 0.75,
+        })}
+      >
+        <Typography variant="microText">
+          {isMac ? "Cmd" : "Ctrl"} + P
+        </Typography>
+      </Box>
+    </Box>
+  );
+};
+
+export const SearchInput: React.FC<{
+  displayedQuery: string;
+  isMobile: boolean;
+  setResultListVisible: (visible: boolean) => void;
+  setQueryText: (queryText: string) => void;
+}> = ({ displayedQuery, isMobile, setResultListVisible, setQueryText }) => {
+  const isMac = navigator.userAgent.toUpperCase().includes("MAC");
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useKeys(["ControlLeft", "KeyP"], (event) => {
+    event.preventDefault();
+    if (!isMac) {
+      inputRef.current?.focus();
+    }
+  });
+
+  useEffect(() => {
+    function checkSearchKey(event: KeyboardEvent) {
+      if (isMac && event.key === "p" && event.metaKey) {
+        event.preventDefault();
+        inputRef.current?.focus();
+      }
+    }
+
+    document.addEventListener("keydown", checkSearchKey);
+
+    return () => {
+      document.removeEventListener("keydown", checkSearchKey);
+    };
+  }, [isMac]);
+
+  const clearSearch = useCallback(() => {
+    setQueryText("");
+  }, [setQueryText]);
+
+  return (
+    <Box
+      sx={(theme) => ({
+        position: "relative",
+        borderRadius: "6px",
+        borderWidth: 1,
+        borderStyle: "solid",
+        borderColor: theme.palette.gray[30],
+        width: isMobile ? "100%" : "385px",
+        ":focus-within": {
+          margin: "-1px",
+          borderWidth: 2,
+          borderColor: theme.palette.blue[70],
+          transition: theme.transitions.create([
+            "border-color",
+            "border-width",
+            "margin",
+          ]),
+          ".MuiInputAdornment-root": {
+            marginRight: "-2px",
+            transition: theme.transitions.create(["margin"]),
+          },
+        },
+      })}
+    >
+      <Box
+        sx={(theme) => ({
+          padding: theme.spacing(0, 1.5),
+          height: "100%",
+          position: "absolute",
+          pointerEvents: "none",
+          display: "flex",
+          alignItems: "center",
+          justifyContent: "center",
+        })}
+      >
+        <SearchIcon sx={{ height: "16px", width: "auto" }} />
+      </Box>
```

Comment:
Just noticed, shouldn't this be a `startAdornment` of the input component?

Reply:
Makes sense. Moved this to `startAdornment` here: https://github.com/hashintel/hash/pull/375/commits/02e021f3f9db8cfb81be506715de82347fe3afcc#diff-ebccff6a5e8b22ec9fa2bd595f473d4b51efdf431feaf92d85f3fc3fd6adcccbR151